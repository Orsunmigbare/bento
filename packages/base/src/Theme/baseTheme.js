import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
