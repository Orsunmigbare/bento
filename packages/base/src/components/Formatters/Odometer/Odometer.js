import React from "react"
import { number } from "prop-types"
import styled from "styled-components"
import Digit from "./Digit"
import hideVisually from "../../../Styles/hideVisually"

class Odometer extends React.PureComponent {
  constructor(...props) {
    super(...props)

    this.state = {
      isAnimating: false
    }

    this.timeout = null

    this.animate = this.animate.bind(this)
  }

  componentWillMount() {
    this.animate()
  }

  componentWillReceiveProps(nextProps) {
    if (this.props.number !== nextProps.number) {
      this.animate()
    }
  }

  animate() {
    this.setState({ isAnimating: true })
    clearTimeout(this.timeout)
    this.timeout = setTimeout(() => {
      this.setState({ isAnimating: false })
    }, this.props.speed + 10)
  }

  render() {
    const { number, speed, size, separatorSteps, ...restProps } = this.props
    const { isAnimating } = this.state
    const chars = parseInt(number, 10)
      .toString()
      .replace("-", "")
      .split("")
    const digitLength = chars.length
    const numSeparators = digitLength / separatorSteps
    for (let i = 0; i < numSeparators; i++) {
      // TODO: There HAS to be a smarter way of doing this. Brain is not working atm
      chars.splice(digitLength - separatorSteps * i, 0, "")
    }

    const renderDigits = (digit, single, speed) =>
      chars.map((digit, i) => (
        <OdometerContainer key={`digit-${i}`}>
          <Digit size={size} digit={digit} single={single} speed={speed} />
        </OdometerContainer>
      ))

    return (
      <OdometerWrapper size={size} {...restProps}>
        <OdometerStatic isAnimating={isAnimating}>
          {renderDigits(number, true, speed)}
        </OdometerStatic>
        <OdometerAnimating isAnimating={isAnimating}>
          {renderDigits(number, false, speed)}
        </OdometerAnimating>
      </OdometerWrapper>
    )
  }
}

const OdometerAnimating = styled.div`
  ${p => (!p.isAnimating ? hideVisually : "")};
  display: flex;
`
const OdometerStatic = styled.div`
  ${p => (p.isAnimating ? hideVisually : "")};
  display: flex;
`

const OdometerContainer = styled.div`
  position: relative;
`

const OdometerWrapper = styled.div`
  font-size: ${p => p.size}px;
  line-height: 1;
  display: inline-block;
`

// TODO: Add support for strings?
Odometer.propTypes = {
  number: number.isRequired,
  separatorSteps: number,
  speed: number,
  size: number.isRequired
}

Odometer.defaultProps = {
  separatorSteps: 3,
  speed: 500
}

export default Odometer
