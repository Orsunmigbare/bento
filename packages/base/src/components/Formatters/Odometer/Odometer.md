```jsx harmony
  <Odometer number={1234567890} speed={50000} size={32}/>
```
