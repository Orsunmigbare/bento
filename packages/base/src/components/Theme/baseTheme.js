export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
