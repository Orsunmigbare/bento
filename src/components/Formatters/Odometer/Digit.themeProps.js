import { themePropTypes } from "../../../theming"

const themeProps = {
  wrapper: {
    name: "digit",
    description: "Wrapper style",
    type: themePropTypes.style
  },
  container: {
    name: "digit_container",
    description: "Container style",
    type: themePropTypes.style
  }
}
export default themeProps
