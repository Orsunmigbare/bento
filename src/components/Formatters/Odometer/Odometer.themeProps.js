import { themePropTypes, createVariantsFunctionFactory } from "../../../theming"

const themeProps = {
  animating: {
    name: "odometer_animating",
    description: "Style for the animating odometer",
    type: themePropTypes.style
  },
  static: {
    name: "odometer_static",
    description: "Style for the static odometer",
    type: themePropTypes.style
  },
  container: {
    name: "odometer_container",
    description: "Style for the container",
    type: themePropTypes.style
  },
  wrapper: {
    name: "odometer_wrapper",
    description: "Style for the wrapper",
    type: themePropTypes.style
  }
}

export default createVariantsFunctionFactory(themeProps)
