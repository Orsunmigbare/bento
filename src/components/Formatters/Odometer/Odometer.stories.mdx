import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Odometer from "./Odometer.js"

<Meta title="Components|Formatters/Odometer" component={Odometer}/>

# Odometer
Displays numbers, and when they change it rolls up to the new value


```javascript
import {Odometer} from "@staccx/bento"
```

## Odometer with number
<Preview>
  <Story name="Default">
   <Odometer number={1000}/>
  </Story>
</Preview>

## Odometer with high number
<Preview>
  <Story name="High number">
   <Odometer number={10000000}/>
  </Story>
</Preview>

## Odometer with low Number
<Preview>
  <Story name="Low number">
   <Odometer number={1}/>
  </Story>
</Preview>

## Odometer with changed seperator steps
<Preview>
  <Story name="SeparatorSteps: 5">
   <Odometer number={100000} separatorSteps={5}/>
  </Story>
</Preview>

## Odometer with low size
<Preview>
  <Story name="Low size">
   <Odometer number={100000} size={5}/>
  </Story>
</Preview>

## Odometer with large size
<Preview>
  <Story name="Large size">
   <Odometer number={100000} size={80}/>
  </Story>
</Preview>



## Props for Odometer
<Props of={Odometer}/>
