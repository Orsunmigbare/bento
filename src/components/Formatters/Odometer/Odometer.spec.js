import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../theming/themes/baseTheme"
import Odometer from "./Odometer"

describe("Odometer", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Odometer />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("Number", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Odometer number={10000} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("seperator steps", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Odometer number={100000} separatorSteps={5} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("Size", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Odometer number={100000} separatorSteps={5} size={80} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("invalid number", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Odometer number="100" separatorSteps={5} />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("invalid separatorSteps", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Odometer number={100000} separatorSteps="5" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
