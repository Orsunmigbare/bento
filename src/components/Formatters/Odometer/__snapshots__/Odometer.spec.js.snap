// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Odometer Snapshots No props 1`] = `
<div
  className="sc-AykKH fIueVJ"
  size={32}
>
  <div
    className="sc-AykKF bsqvYw"
  >
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC dNUxlM"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-AykKE nmeVQ"
  >
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD btghud"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC dNUxlM"
        size={32}
      >
        <div
          className="sc-AykKD jEkgQB"
          size={32}
          speed={500}
        >
          
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Odometer Snapshots Number 1`] = `
<div
  className="sc-AykKH fIueVJ"
  size={32}
>
  <div
    className="sc-AykKF bsqvYw"
  >
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          1
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC dNUxlM"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC dNUxlM"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-AykKE nmeVQ"
  >
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD jaOeyL"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD btghud"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC dNUxlM"
        size={32}
      >
        <div
          className="sc-AykKD jEkgQB"
          size={32}
          speed={500}
        >
          
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD btghud"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD btghud"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD btghud"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC dNUxlM"
        size={32}
      >
        <div
          className="sc-AykKD jEkgQB"
          size={32}
          speed={500}
        >
          
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Odometer Snapshots Size 1`] = `
<div
  className="sc-AykKH eCyJwk"
  size={80}
>
  <div
    className="sc-AykKF bsqvYw"
  >
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC gbWNzG"
        size={80}
      >
        <div
          className="sc-AykKD heuYLW"
          size={80}
          speed={500}
        >
          1
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC imcNnE"
        size={80}
      >
        <div
          className="sc-AykKD heuYLW"
          size={80}
          speed={500}
        >
          
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC gbWNzG"
        size={80}
      >
        <div
          className="sc-AykKD heuYLW"
          size={80}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC gbWNzG"
        size={80}
      >
        <div
          className="sc-AykKD heuYLW"
          size={80}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC gbWNzG"
        size={80}
      >
        <div
          className="sc-AykKD heuYLW"
          size={80}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC gbWNzG"
        size={80}
      >
        <div
          className="sc-AykKD heuYLW"
          size={80}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC gbWNzG"
        size={80}
      >
        <div
          className="sc-AykKD heuYLW"
          size={80}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC imcNnE"
        size={80}
      >
        <div
          className="sc-AykKD heuYLW"
          size={80}
          speed={500}
        >
          
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-AykKE nmeVQ"
  >
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC gbWNzG"
        size={80}
      >
        <div
          className="sc-AykKD bwFncq"
          size={80}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC imcNnE"
        size={80}
      >
        <div
          className="sc-AykKD fzpUgU"
          size={80}
          speed={500}
        >
          
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC gbWNzG"
        size={80}
      >
        <div
          className="sc-AykKD fesZwE"
          size={80}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC gbWNzG"
        size={80}
      >
        <div
          className="sc-AykKD fesZwE"
          size={80}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC gbWNzG"
        size={80}
      >
        <div
          className="sc-AykKD fesZwE"
          size={80}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC gbWNzG"
        size={80}
      >
        <div
          className="sc-AykKD fesZwE"
          size={80}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC gbWNzG"
        size={80}
      >
        <div
          className="sc-AykKD fesZwE"
          size={80}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC imcNnE"
        size={80}
      >
        <div
          className="sc-AykKD fzpUgU"
          size={80}
          speed={500}
        >
          
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Odometer Snapshots invalid number 1`] = `
<div
  className="sc-AykKH fIueVJ"
  size={32}
>
  <div
    className="sc-AykKF bsqvYw"
  >
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          1
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC dNUxlM"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-AykKE nmeVQ"
  >
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD jaOeyL"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD btghud"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD btghud"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC dNUxlM"
        size={32}
      >
        <div
          className="sc-AykKD jEkgQB"
          size={32}
          speed={500}
        >
          
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Odometer Snapshots invalid separatorSteps 1`] = `
<div
  className="sc-AykKH fIueVJ"
  size={32}
>
  <div
    className="sc-AykKF bsqvYw"
  >
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          1
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC dNUxlM"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC dNUxlM"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-AykKE nmeVQ"
  >
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD jaOeyL"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC dNUxlM"
        size={32}
      >
        <div
          className="sc-AykKD jEkgQB"
          size={32}
          speed={500}
        >
          
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD btghud"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD btghud"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD btghud"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD btghud"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD btghud"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC dNUxlM"
        size={32}
      >
        <div
          className="sc-AykKD jEkgQB"
          size={32}
          speed={500}
        >
          
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Odometer Snapshots seperator steps 1`] = `
<div
  className="sc-AykKH fIueVJ"
  size={32}
>
  <div
    className="sc-AykKF bsqvYw"
  >
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          1
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC dNUxlM"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC dNUxlM"
        size={32}
      >
        <div
          className="sc-AykKD mFyTn"
          size={32}
          speed={500}
        >
          
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-AykKE nmeVQ"
  >
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD jaOeyL"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC dNUxlM"
        size={32}
      >
        <div
          className="sc-AykKD jEkgQB"
          size={32}
          speed={500}
        >
          
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD btghud"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD btghud"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD btghud"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD btghud"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC iFSbYa"
        size={32}
      >
        <div
          className="sc-AykKD btghud"
          size={32}
          speed={500}
        >
          0 1 2 3 4 5 6 7 8 9 0
        </div>
      </div>
    </div>
    <div
      className="sc-AykKG hivkXU"
    >
      <div
        className="sc-AykKC dNUxlM"
        size={32}
      >
        <div
          className="sc-AykKD jEkgQB"
          size={32}
          speed={500}
        >
          
        </div>
      </div>
    </div>
  </div>
</div>
`;
